scripts/hand_test.py

```diff
@@ -1,81 +1,99 @@
+                               queue_size=1)
 rate = rospy.Rate(100)
 
-def close_hand():
-    hand_msg.setpoint = [900]*len(hand_msg.global_id)
-    motorcmd_pub.publish(hand_msg)
-
-def open_hand():
-    hand_msg.setpoint = [0]*len(hand_msg.global_id)
-    motorcmd_pub.publish(hand_msg)
-
-def move_fingers(right_hand = True, targets = [0,0,0,0]):
+def move_fingers(right_hand=True, targets=None):
     """
-    :param target: array of finger target values between 0 (open hand) and 800 (closed hand)
-    lr: "left" or "right"
+    :param target: array of finger target values between 0 (open hand) and 1 (closed hand)
+    right_hand: True (right hand); False (left hand)
     left_ids = [38, 39, 40, 41]
     right_ids = [42, 43, 44, 45]
     """
+    if targets is None:
+        targets = [0, 0, 0, 0]
 
+    # just take the ids with actual values (>0)
+    right_ids = [42, 42, 44, 45]
+    left_ids = [38, 39, 40, 41]
     if right_hand:
-        hand_msg.global_id = [42, 42, 44, 45]
-
+        ids_rl = right_ids
     else:
-        hand_msg.global_id = [38, 39, 40, 41]
+        ids_rl = left_ids
+    targets_relevant = []
+    ids_picked = []
 
-    hand_msg.setpoint = [i*800 for i in targets]
+    for j in range(len(targets)):  # check if there are 0- values -> motor is not commanded
+        if targets[j] > 0.0:
+            ids_picked.append(ids_rl[j])
+            targets_relevant.append(targets[j])
+    hand_msg.global_id = ids_picked
+
+    # make sure values are within the limits
+    targets = clamp_values(targets_relevant, min_allowed_value=0.0, max_allowed_value=1.0)
+    hand_msg.setpoint = [i * 800 for i in targets]
     motorcmd_pub.publish(hand_msg)
-    rospy.sleep(0.07)
 
-def move_fingers_simultaneous(right_hand = True):
-    # hand finger moving test all fingers move the same
+def clamp_values(targets, min_allowed_value, max_allowed_value):
+    for t in range(len(targets)):
+        if targets[t] < min_allowed_value:
+            targets[t] = min_allowed_value
+            print("Warning: target value too low -> clamped it")
+        elif targets[t] > max_allowed_value:
+            targets[t] = max_allowed_value
+            print("Warning: target value too high -> clamped it")
+    return targets
+
+
+def move_pointer_finger(right_hand=True, intervals=11):
     ranges = [0, 1]
-    samples = 101
 
     for i in range(2):
-        if not i%2:
-            print(right_hand, "close")
-            target_value = np.linspace(ranges[0], ranges[1], samples)  # start with 0 (open) -> 1 (close)
+        if not i % 2:
+            print(right_hand, "pointer finger close")
+            target_value = np.linspace(ranges[0], ranges[1], intervals)  # start with 0 (open) -> 1 (close)
             for pos in target_value:
-                targets = [pos] * len(hand_msg.global_id)  # put the current pos value in a 4-element list
+                targets = [0, pos, pos,
```

Comment:
so as i wrote before, when you want to move a single finger, publish a motor command to a single id. you message will look like this:

```python 
msg = MotorCommand()
msg.setpoint = [800]
msg.global_id = [42]
publisher.publish(msg)
```